### PR TITLE
Move parser into its own package

### LINKED ISSUE #40: Pull parser.go into a separate package
To implement https://github.com/containous/staert/issues/35 and reuse the parsing code, it will be good to reuse flaeg parser, as suggested for a previous tentative : https://traefik.slackarchive.io/dev/page-17/ts-1497598028679523

This requires quite a lot of renaming and prefixing in the code, so I'd like to make sure I plan this right

- Suggested package name : `parse`
- Keep it under flaeg, not a different repo
- Move `parser.go` and `parser_test.go` into a `parse` folder. rename files to `parse.go/parse_test.go`
- Import `github.com/containous/flaeg/parse` in `flaeg` 
- Make all types in parse public
- Prefix every call to parse types by `parse.`
- Same in test files
- There is a lot of repetition in the flaeg tests, maybe look into refactoring 

Sounds good?

### PR DESCRIPTION
**Pull request by serbrech**
Fixes #40 (minus potential refactoring)

~Please review in particular the `Duration` it was a bit more tricky due to the naming ambiguity between `time.Duration` and `parse.Duration` should be alright though as all tests pass~

### CODE REVIEW

**Comment by serbrech** [comment]
I made a mistake early with Duration and changed `Duration` to time.Duration in `parse.go`
that lead me to wrongly fixing compile and test error in parse_test and flaeg, until everything worked... but with time.Duration. After a chat with @ldez, I found the root of my mistake and reverted to use parse.Duration where it was changed in e1a5fb9

File: flaeg.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ import (
 	"text/template"
 	"time"
 
+	parse "github.com/containous/flaeg/parse"
+
```
Comment (ldez) [inline flaeg.go]: could you remove this empty line.

Comment (ldez) [reply flaeg.go]: and remove the alias

Comment (serbrech) [reply flaeg.go]: Do you use golint? 
Any specific configuration? 
I get a few warnings regarding comments and error message formats.

**Comment by ldez** [comment]
Travis don't want to build this PR... 😞 
I'm going closed and reopened.

**Comment by ldez** [comment]
Don't work... 😞 
I will try to make a dummy commit.

**Comment by ldez** [comment]
I tested the PR locally, I bypass Travis.

**Pull request merged**